test/instrumentation/modules/mysql2/mysql.js

```diff
@@ -0,0 +1,566 @@
+'use strict'
+
+var host = process.env.MYSQL_HOST
+var agent = require('../../../..').start({
+  appName: 'test',
+  secretToken: 'test',
+  captureExceptions: false
+})
+
+var test = require('tape')
+var semver = require('semver')
+var mysql = require('mysql2')
+var mysqlVersion = require('mysql/package.json').version
+var utils = require('./_utils')
+
+var queryable
+var factories = [
+  [createConnection, 'connection'],
+  [createPool, 'pool'],
+  [createPoolAndGetConnection, 'pool > connection'],
+  [createPoolClusterAndGetConnection, 'poolCluster > connection'],
+  [createPoolClusterAndGetConnectionViaOf, 'poolCluster > of > connection']
+]
+
+factories.forEach(function (f) {
+  var factory = f[0]
+  var type = f[1]
+
+  // Prior to mysql v2.4 pool.query would not return a Query object.
+  // See: https://github.com/mysqljs/mysql/pull/830
+  // var skipStreamTest = type === 'pool' && semver.satisfies(mysqlVersion, '<2.4.0')
+  var skipStreamTest = false
+  // Prior to mysql v2.2 pool.query required a callback.
+  // See: https://github.com/mysqljs/mysql/pull/585
+  var skipNoCallbackTest = type === 'pool' && semver.satisfies(mysqlVersion, '<2.2.0')
+  skipNoCallbackTest = true
```

Comment:
This boolean is probably not needed for mysql2. If that's the case it's better if we just remove it as it makes it easier to read the code 😃